cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -52,38 +52,64 @@ func (u *Upgrade) CanUpgradeKubelets() bool {
 	return !sameVersionFound
 }
 
-// CanUpgradeEtcd returns whether an upgrade of etcd is possible
-func (u *Upgrade) CanUpgradeEtcd() bool {
-	return u.Before.EtcdVersion != u.After.EtcdVersion
-}
-
-// ClusterState describes the state of certain versions for a cluster
+// ClusterState describes the state of certain versions for a cluster during an upgrade
 type ClusterState struct {
 	// KubeVersion describes the version of the Kubernetes API Server, Controller Manager, Scheduler and Proxy.
 	KubeVersion string
```

Comment:
should this be changed as we discussed here:
https://github.com/kubernetes/kubernetes/pull/123578/files#r1508641110

at least the comment
```suggestion
	// KubeVersion describes the version of latest Kubernetes API Server in the cluster.
	KubeVersion string
```